packages/scandipwa/src/component/RenderWhenVisible/RenderWhenVisible.component.js

```diff
@@ -52,16 +51,31 @@ export class RenderWhenVisible extends PureComponent {
         }
     };
 
-    shouldRender(isVisible) {
-        return isVisible || this.wasVisible;
+    shouldRender() {
+        const { wasVisible } = this.state;
+        return !wasVisible && !this.isSSR() && !this.isCrawler();
     }
 
     handleVisibilityToggle = (isVisible) => {
-        if (!this.wasVisible && isVisible) {
+        const { wasVisible } = this.state;
+
+        if (!wasVisible && isVisible) {
             this.setState({ wasVisible: true });
         }
     };
 
+    isSSR() {
+        return !globalThis.window || !globalThis.window.document;
+    }
```

Comment:
Same as below - consider making this an util.

Reply:
Done